src/raft_log.c

```diff
@@ -61,16 +55,18 @@ static int __ensurecapacity(log_private_t * me, int n)
         return RAFT_ERR_NOMEM;
 
 
-    if (me->front < me->back)
+    if (me->front + me->count <= me->size)
     {
         memcpy(&temp[0], &me->entries[me->front],
-               sizeof(raft_entry_t) * (me->back - me->front));
+               me->count * sizeof(raft_entry_t));
     }
     else if (me->count > 0)
     {
-        int i = me->size - me->front;
-        memcpy(&temp[0], &me->entries[me->front], i * sizeof(raft_entry_t));
-        memcpy(&temp[i], &me->entries[0], me->back * sizeof(raft_entry_t));
+        int index = me->size - me->front;
+        memcpy(&temp[0], &me->entries[me->front],
+               index * sizeof(raft_entry_t));
+        memcpy(&temp[index], &me->entries[0],
+               (me->count - index) * sizeof(raft_entry_t));
     }
```

Comment:
How about:
```
if (0 < me->count) {
    int n = me->size - me->front;
    if (me->count <= n)
    {
        memcpy(&temp[0], &me->entries[me->front], sizeof(raft_entry_t) * me->count);
    }
    else
    {
        memcpy(&temp[0], &me->entries[me->front], sizeof(raft_entry_t) * n);
        memcpy(&temp[n], &me->entries[0], sizeof(raft_entry_t) * (me->count - n));
    }
}
```

Reply:
Errrh, I'm not a fan of this style, but:
```
if (0 < me->count)
{
```

Reply:
Is there any advantage to doing it this way?

Reply:
Never mind.. I see why you suggested it this way.

Reply:
It was only about readability; the previous code should work just fine. (Notice I approved the previous version.) The "count > 0" check gives a clue that we are copying existing entries and prevent calling memcpy to copy 0 bytes. The n variable has the same meaning as the index variable in the previous code, but I think the value is more a number of entries than an index. Sorry for being picky.